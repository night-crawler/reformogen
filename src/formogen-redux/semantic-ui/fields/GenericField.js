import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { GenericField as GenericFieldComponent } from '../../../formogen/semantic-ui/fields/GenericField';

// import { } from '../../selectors';
// import { bootstrap } from '../../actions';


function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

function mergeProps(stateProps, dispatchProps, ownProps) {
  const { dispatch } = dispatchProps;

  return {
    ...ownProps,
    ...stateProps,
    ...dispatchProps,

    dispatch: undefined,
  };
}

export const GenericField = connect(
  createStructuredSelector({

  }),
  mapDispatchToProps,
  mergeProps,
)(GenericFieldComponent);
