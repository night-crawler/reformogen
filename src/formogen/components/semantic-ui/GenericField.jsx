import React from 'react';
import loglevel from 'loglevel';
import { Form } from 'semantic-ui-react';
import _ from 'lodash';

import { MessageList } from './MiscComponents';

import Label from './Label';

import propTypes from '../fieldPropTypes';


/**
 * Generic Field Component to use in DEBUG MODE ONLY
 * @param {Object} props
 * @returns {XML}
 * @constructor
 */
GenericField.propTypes = propTypes;
export default function GenericField(props) {
    loglevel
        .getLogger('Formogen/components/semantic-ui/GenericField.jsx')
        .warn(`Using a placeholder GenericFieldComponent to render field "${props.name}" of type "${props.type}"`);

    let _props = Object.assign({}, props);
    if (_.isFunction(props.updateProps)) {
        _props = props.updateProps(_props, props);
    }

    return (
        <Form.Field
            required={ props.required }
            disabled={ !props.editable }
            width={ props.layoutOpts.width }
            error={ !_.isEmpty(props.errors) }
        >
            <Label { ...props } />
            <pre>
                { JSON.stringify(_props, null, 4) }
            </pre>
            { !props.helpTextOnHover ? <span className="help-text">{ props.help_text }</span> : ''}
            <MessageList messages={ props.errors }/>
        </Form.Field>
    );
}
