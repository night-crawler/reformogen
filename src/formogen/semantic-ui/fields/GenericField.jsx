import React from 'react';
import _ from 'lodash';
import { Form } from 'semantic-ui-react';

import propTypes from '../../fieldPropTypes';
import { FieldLabel } from '../common/FieldLabel';
import { ErrorsList } from '../common/ErrorsList';


/**
 * Generic Field Component to use in DEBUG MODE ONLY
 * @param {Object} props
 * @returns {XML}
 * @constructor
 */
GenericField.propTypes = propTypes;
export function GenericField(props) {
  // eslint-disable-next-line
  console.warn(`Using a placeholder GenericFieldComponent to render field "${ props.name }" of type "${ props.type }"`);

  let _props = Object.assign({}, props);
  if (_.isFunction(props.updateProps)) {
    _props = props.updateProps(_props, props);
  }

  return (
    <Form.Field
      required={ props.required }
      disabled={ !props.editable }
      width={ props.layoutOpts.width }
      error={ !_.isEmpty(props.errors) }
    >
      <FieldLabel { ...props } />
      <pre>
        { JSON.stringify(_props, null, 4) }
      </pre>
      { !props.helpTextOnHover
        ? <span className='help-text'>{ props.help_text }</span>
        : ''
      }
      <ErrorsList messages={ props.errors } />
    </Form.Field>
  );
}
