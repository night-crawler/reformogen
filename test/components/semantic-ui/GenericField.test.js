import GenericField from 'components/semantic-ui/GenericField';


it('should render GenericField', () => {
    const wrapper = shallow(
        <GenericField
            type='LOLField'
            name='something_cute'
            verbose_name='Die die die my darling'
            help_text="Don't utter a single word"
            layoutOpts={ {width: 1} }
        />
    );
    expect(wrapper).toMatchSnapshot();
});
