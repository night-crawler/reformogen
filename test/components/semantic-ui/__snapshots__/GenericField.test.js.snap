// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GenericField 1`] = `
<FormField
  disabled={true}
  error={false}
  width={1}
>
  <SUILabel
    help_text="Don't utter a single word"
    layoutOpts={
      Object {
        "width": 1,
      }
    }
    name="something_cute"
    type="GenericField"
    verbose_name="Die die die my darling"
  />
  <pre>
    {
    "type": "GenericField",
    "name": "something_cute",
    "verbose_name": "Die die die my darling",
    "help_text": "Don't utter a single word",
    "layoutOpts": {
        "width": 1
    }
}
  </pre>
  <span
    className="help-text"
  >
    Don't utter a single word
  </span>
  <MessageList />
</FormField>
`;
